### ISSUE TITLE
If i create 3 three slide, all 3 slides fit into one slide and remaining slide are empty

### ISSUE THREAD

**Issue by karna419** [nolimits4web/swiper#3194]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* Platform/Target and Browser Versions: PLATFORM CLIENT YOU ARE TARGETING SUCH AS macOS, Windows, CORDOVA, IOS, ANDROID, CHROME, ETC.
* Live Link or JSFiddle/Codepen or website with isssue: PREFERABLY _(IF YOU WANT YOUR ISSUE TO BE RESOLVED ASAP)_.

### What you did
I am using three tabs and in last tab i want to use slides for its content.

<div class="section">

                <div class="swiper-container-main">
                    <div class="swiper-wrapper">
                        <div class="swiper-slide" style="height:0%;" id="slide1">
                            <div class="contact_div">
                                <div class="contact_top">
                                    <div class="contact_chk">
                                        <input id="fb_check_box" type="checkbox" class="contact_fb check_txt" name="contacts" value="facebook friends" checked>
                                        <span class="chk_label">Facebook Friends</span>
                                    </div>
                                    <div class="contact_chk">
                                        <input id="mm_check_box"type="checkbox" class="contact_mm check_txt" name="contacts" value="meet market friends" checked>
                                        <span class="chk_label">Meet Market Friends</span>
                                    </div>
                                </div>
                                <div class="contact_btm">
                                    <div class="alphabet_div"  id="contact_btn"></div>
                                    <div class="alphabet_left" ></div>
                                    <div class="alphabet_right" ></div>
                                </div>
                            </div>
                        </div>
                        <div class="swiper-slide" style="height:0%;" id="slide2">
                            <div class="message_div" id="chat">
                                <div class="msg_ls" onClick=initPrivateChat()>

                                </div>
                            </div>
                        </div>
                        <div class="swiper-slide" id="slide3">
                            <div class="event_div">
                                <div class="event_top">
                                    <div class="saved_evtdiv">
                                <span class="savedEvents_span">SAVED EVENTS
                                    <span id="savedEvents_txt">0</span>
                                </span>

                                    </div>
                                    <div class="filter_seldiv">
                                        <span class="filter_sel filter_span">FILTER EVENTS</span>
                                    </div>
                                </div>

                                <div class="event_section">
                                    <div class="swiper-container"><div class="swiper-wrapper" id="display-resources"></div></div>
                                </div>
                                <div class="event_footer">
                                    <input type="button" class="create_btn" value="CREATE EVENT">
                                    <input type="button" class="my_event_btn" style="margin-left:1%;" value="My Events">
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
In my js file, i never used swiper initiation inside deviceready. But i initiated swiper in a function that was called from deviceready. In my case, As i said earlier, I am using parent swipe for 3 tabs and in 3rd tabs i am using another swiper for it's content. Although parent swiper-container is working, child swiper container is not working. The all the number slide's content fits into first slide (child container) and remaining slides placeholder is showing empty. The js code are 
                     var mySwiper = $('.swiper-container').swiper({
                       speed: 900,
                     });
var slides = [];
//variable slides is not empty in actual code and i am using array of string for slides.
mySwiper.appendSlide(slides);
                    //duplicate slides for just demo
                    $.each(slides, function(index, slide){
                          mySwiper.appendSlide(slide);
                    })
                    mySwiper.update();

### Expected Behavior
I need a normal slide view

### Actual Behavior
All slides are coming into first slide layout and remaining shows as empty.

**Comment by karna419** [comment]
I fixed the issue by setting up width for a slide and setting up width from multiplying total slides width in the child container. Most examples i have seen on internet shows swiper-container width is setup for single slide. but in my case i had to set container width by multiplying slides width. Is that wright way?. Also after fixing this issues, i am facing another. Issue is when i slide forward, next slide is not coming to the centre place easy, barely it's working. I had to swipe multiple times to make the next slide come to centre place. If move backward , directly comes to first slide.
